pkg/pmem-csi-operator/controller/deployment/controller_driver.go

```diff
@@ -179,18 +203,29 @@ func (d *PmemCSIDriver) reconcileDeploymentChanges(r *ReconcileDeployment, exist
 		}
 	}
 
-	if updateController {
-		klog.Infof("Updating controller driver for deployment %q", d.Name)
-		if e := r.Update(d.getControllerStatefulSet()); e != nil {
-			requeue = true
-			err = e
+	// Force update all deployment objects
+	if updateAll {
+		klog.Infof("Updating all objects for deployment %q", d.Name)
+		for _, obj := range d.getDeploymentObjects(r) {
+			if e := r.Update(obj); e != nil {
```

Comment:
This doesn't update the secrets. Let's merge https://github.com/intel/pmem-csi/pull/586 and the `getDeploymentObjects` truly returns *all* objects.

Reply:
Another thought: what happens if the object doesn't exist because the operator we update from didn't create it?

Reply:
Secrets update is a bit tricky, With the current code, for a deployment with no keys/certificates set, It creates a new set of secrets for every call.

Strictly speaking of this function, it only deals with the objects we created in the Initializing phase, so this will not/need not update the secrets.

Reply:
I don't follow. We need to update all objects only if we found an existing deployment in an unknown state. Why would that then get triggered for every call of the function?

Reply:
We end up in this function, only when the deployment is in "Running" phase. So for that deployment secretes must have already created and needs no update. And moreover, secrets creation is not idempotent in case of a deployment using operator generated keys/certificates.

Reply:
> We end up in this function, only when the deployment is in "Running" phase.

Which is a problem. We need to update stale objects regardless of the phase.

Let's do this:
- review and merge https://github.com/intel/pmem-csi/pull/586 because it simplifies where we create objects
- remove the  "operator: reload state of a running deployment if not found in cache" from this PR because the rest should be okay
- write E2E test cases for reconciling on startup, then check whether the new code passes those tests